content/articles/spatial_mgrs/index.Rmd

```diff
@@ -0,0 +1,427 @@
+The MGRS specification is maintained by the NGA Office of Geomatics ^[URL: <https://earth-info.nga.mil>.] [@national_geospatial-intelligence_agency_universal_2014].
+Best use the foregoing reference when referring to the MGRS.
+Other resources describe the MGRS too, including @wikipedia_contributors_military_2023.
+
+Biodiversity monitoring and perhaps other projects sometimes claim to use UTM grid references, while they actually use MGRS grid references.
+Hence it is good to understand the difference; see also the [Terminological notes](#terminological-notes).
+
+The coordinate system obtained by **UTM** projection is used in the MGRS definition between 80°S and 84°N, and MGRS combines this with a **UPS** (Universal Polar Stereographic) projection to define its grid in the polar regions.
+In what follows, _only the UTM portion of MGRS_ is considered.
+
+The MGRS grid pattern between 80°S and 84°N has two important differences with the pattern of the [UTM grid](#utm-grid)—apart from using a different referencing (geocoding) system:
+
+- In MGRS the UTM grid cell pattern is _intersected_ by latitude bands of generally 8° wide (starting at the equator).
+Intersecting [UTM zones](#utm-projection) with latitude bands results in large cells called _**MGRS grid zones**_ of generally 6° longitude by 8° latitude large.
+- In MGRS, some of the MGRS grid zones are wider or narrower than 6 longitudinal degrees.
+This has been done to avoid grid zone junctions in specific areas, e.g. Bergen and Oslo (Norway) belong to the same grid zone.
+
+```{r utm-grid-latbands-europe, fig.cap="The intersection of the UTM grid by latitude bands of 8° wide forms the basis of the MGRS. This map does not include the subsequent changes to some MGRS grid zones in the north, e.g. in Norway."}
+ggplot(data = countries) +
+  geom_sf(fill = "grey97", colour = "grey50") + 
+  grids$geom_3035[1:14] +
+  coord_sf(datum = "EPSG:4326", xlim = c(2.5e6, 6e6), ylim = c(1.5e6, 4.5e6)) +
+  scale_x_continuous(breaks = seq(-24, 48, 6)) +
+  scale_y_continuous(breaks = seq(0, 80, 8)) +
+  theme(panel.grid = element_line(colour = "blue"))
+```
+
+Below the grid zone level it is still the UTM grid pattern that drives the further subdivision in MGRS between 80°S and 84°N, down to the required precision, just as in other grid systems.
+However an important difference at this level is that MGRS uses an _**alphanumeric grid reference**_ instead of a numeric one as in civil grids.
+In MGRS, Latin characters are used in referencing both the applicable grid zone and the first subdivision thereof, as follows.
+
+An MGRS 'grid reference' (sometimes called MGRS coordinate) is defined as:
+
+```
+[Grid Zone Designation] [Grid Square ID] [Truncated integer X-coordinate] [Truncated integer Y-coordinate]
+```
+
+An example is `31U ES 56914 61249` at the 1 m precision, which resolves to `31U ES 56 61` at 1 km precision.
+
```

Comment:
And this explains why the spaces can be omitted in a computational environment?

Reply:
> I suggest using the same example as used in the UTM grid section

Yes, good idea.

> I get the impression that MGRS grid always has same number of digits for the truncated integer X and Y? Whereas for UTM grid this is not the case in general?

You raised a good point @hansvancalster. This was not yet well clarified in the text and I hadn't paid much attention to potentially differing numbers of digits between X and Y. I have looked up again what sources say for MGRS and UTM. One important distinction of course, between MGRS and UTM grid referencing, is:

- UTM grid referencing in its full form contains the anterior digits of the UTM coordinates. The UTM X **coordinate** (easting) is constrained by the UTM zone borders, but given the false easting it will always have 6 digits at the meter precision, whereas the UTM Y **coordinate** (northing) can run from value 0 to a number of 7 digits at meter precision. See some R code further below. Truncating these (hence generating UTM grid reference) to a lower precision will not change that difference.
- the numerical part of MGRS grid referencing has a maximum of 5 digits, since it captures only the digits below 100000 m. So in general these numbers always run from 0 m to 99999 m in X and Y direction.

MGRS is strict about the numbers of digits for X and Y: it is 0 to 5 digits and this is directly linked to the precision. This implicitly means adding leading zeros, although this could be pointed out better by NGA. Example: X = 450 m becomes 00450; X = 9 km becomes 09. So indeed here you always have the same number of digits in X and Y.

Hence the omission of _spaces_ in UTM grid referencing is tricky because it potentially leads to combining X and Y with different number of digits. As far as I can see, omission of spaces is only applied in notations that drop anterior digits and in the context of the USA, which means there _are_ anterior digits to be dropped in both X and Y direction (the digits representing 10^5 m and higher). So leading zeros can come out of this operation (e.g. an X coordinate of 400125 will be shortened to 00125 and a Y coordinate 5604109 to 04109). Consequently you get a regular string (X and Y of equal length) in such case. Because there _are_ anterior digits, the operation makes sense if you drop spaces. Close to the equator (below 1° latitude), you will get into trouble if you would drop anterior digits and concatenate.

So basically, if the UTM grid reference is _not_ simplified, some punctuation is applied to separate X and Y as these often have a different number of digits. That is my conclusion here.

``` r
library(sf)
#> Linking to GEOS 3.11.1, GDAL 3.6.4, PROJ 9.1.1; sf_use_s2() is TRUE
utm31N_zonewidth <- function(latitude) {
  st_sfc(st_linestring(matrix(c(0, latitude, 6, latitude), ncol = 2, byrow = TRUE)), crs = 4326) |> 
    st_transform("EPSG:32631") |> 
    st_length()
}
utm31N_zonewidth(latitude = 0)
#> 667957.1 [m]
utm31N_zonewidth(latitude = 51)
#> 420977.7 [m]
utm31N_zonewidth(latitude = 84)
#> 69989.31 [m]

utm31N_project_wgs84 <- function(x, y) {
  st_sfc(st_point(c(x, y)), crs = 4326) |> 
    st_transform("EPSG:32631") |> 
    st_coordinates() |> 
    as.numeric()
}
utm31N_project_wgs84(0, 0)
#> [1] 166021.4      0.0
utm31N_project_wgs84(6, 0)
#> [1] 833978.6      0.0
utm31N_project_wgs84(0, 1)
#> [1] 166072.1 110682.8
utm31N_project_wgs84(6, 1)
#> [1] 833927.9 110682.8
utm31N_project_wgs84(0, 51)
#> [1]  289511.1 5654109.2
utm31N_project_wgs84(6, 51)
#> [1]  710488.9 5654109.2
utm31N_project_wgs84(0, 84)
#> [1]  465005.3 9329005.2
utm31N_project_wgs84(6, 84)
#> [1]  534994.7 9329005.2
```

<sup>Created on 2023-09-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.1 (2023-06-16)
#>  os       Linux Mint 21.2
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language nl_BE:nl
#>  collate  nl_BE.UTF-8
#>  ctype    nl_BE.UTF-8
#>  tz       Europe/Brussels
#>  date     2023-09-25
#>  pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  class         7.3-22  2023-05-03 [3] RSPM (R 4.2.0)
#>  classInt      0.4-10  2023-09-05 [3] RSPM (R 4.3.0)
#>  cli           3.6.1   2023-03-23 [3] RSPM (R 4.2.0)
#>  DBI           1.1.3   2022-06-18 [3] RSPM (R 4.2.0)
#>  digest        0.6.33  2023-07-07 [3] RSPM (R 4.2.0)
#>  dplyr         1.1.3   2023-09-03 [3] RSPM (R 4.3.0)
#>  e1071         1.7-13  2023-02-01 [3] RSPM (R 4.2.0)
#>  evaluate      0.21    2023-05-05 [3] RSPM (R 4.2.0)
#>  fansi         1.0.4   2023-01-22 [3] RSPM (R 4.2.0)
#>  fastmap       1.1.1   2023-02-24 [3] RSPM (R 4.2.0)
#>  fs            1.6.3   2023-07-20 [3] RSPM (R 4.2.0)
#>  generics      0.1.3   2022-07-05 [3] RSPM (R 4.2.0)
#>  glue          1.6.2   2022-02-24 [3] RSPM (R 4.2.0)
#>  htmltools     0.5.6   2023-08-10 [3] RSPM (R 4.2.0)
#>  KernSmooth    2.23-22 2023-07-10 [3] RSPM (R 4.2.0)
#>  knitr         1.44    2023-09-11 [3] RSPM (R 4.3.0)
#>  lifecycle     1.0.3   2022-10-07 [3] RSPM (R 4.2.0)
#>  magrittr      2.0.3   2022-03-30 [3] RSPM (R 4.2.0)
#>  pillar        1.9.0   2023-03-22 [3] RSPM (R 4.2.0)
#>  pkgconfig     2.0.3   2019-09-22 [3] CRAN (R 4.0.1)
#>  proxy         0.4-27  2022-06-09 [3] RSPM (R 4.2.0)
#>  purrr         1.0.2   2023-08-10 [3] RSPM (R 4.2.0)
#>  R.cache       0.16.0  2022-07-21 [3] RSPM (R 4.2.0)
#>  R.methodsS3   1.8.2   2022-06-13 [3] RSPM (R 4.2.0)
#>  R.oo          1.25.0  2022-06-12 [3] RSPM (R 4.2.0)
#>  R.utils       2.12.2  2022-11-11 [3] RSPM (R 4.2.0)
#>  R6            2.5.1   2021-08-19 [3] RSPM (R 4.2.0)
#>  Rcpp          1.0.11  2023-07-06 [3] RSPM (R 4.2.0)
#>  reprex        2.0.2   2022-08-17 [3] RSPM (R 4.2.0)
#>  rlang         1.1.1   2023-04-28 [3] RSPM (R 4.2.0)
#>  rmarkdown     2.25    2023-09-18 [3] RSPM (R 4.3.0)
#>  rstudioapi    0.15.0  2023-07-07 [3] RSPM (R 4.2.0)
#>  sessioninfo   1.2.2   2021-12-06 [3] RSPM (R 4.2.0)
#>  sf          * 1.0-14  2023-07-11 [1] RSPM (R 4.3.1)
#>  styler        1.10.2  2023-08-29 [3] RSPM (R 4.2.0)
#>  tibble        3.2.1   2023-03-20 [3] RSPM (R 4.3.0)
#>  tidyselect    1.2.0   2022-10-10 [3] RSPM (R 4.2.0)
#>  units         0.8-4   2023-09-13 [3] RSPM (R 4.3.0)
#>  utf8          1.2.3   2023-01-31 [3] RSPM (R 4.2.0)
#>  vctrs         0.6.3   2023-06-14 [3] RSPM (R 4.2.0)
#>  withr         2.5.0   2022-03-03 [3] RSPM (R 4.2.0)
#>  xfun          0.40    2023-08-09 [3] RSPM (R 4.2.0)
#>  yaml          2.3.7   2023-01-23 [3] RSPM (R 4.2.0)
#> 
#>  [1] /home/floris/lib/R/library
#>  [2] /usr/local/lib/R/site-library
#>  [3] /usr/lib/R/site-library
#>  [4] /usr/lib/R/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

Reply:
> spaces can be omitted in a computational environment

That is indeed because MGRS has the strict symmetric format, and I guess it is advised because strings without spaces are shorter and a little easier to parse computationally.

NGA is not entirely clear about the 'why', so I'm just guessing actually.

> To facilitate machine-to-machine communication, an MGRS string is to have no intermediate spaces or punctuation
marks and all the letters are to be capitals.

Reply:
Some more info about:

> omission of spaces is only applied in notations that drop anterior digits and in the context of the USA

  Stott (1977) says about the example `18.501260.4592120`:  "_It is this 15-digit form that is at the moment most common in the U.S. [...] It is difficult, however, to determine from the grid reference the approximate location or map name, and there is a variety of established means of including this information and shortening the grid reference._" This is then about dropping zone number and anterior digits, and optionally prefixing the resulting reference with the topographic map name (the quadrangle name) to make it unequivocal, leading to UTM grid notations like e.g. `SHOHOLA 012921` at 100 m precision or `SHOHOLA, PA 01260/92120` at 1 m precision (coordinates separator: point, slash, or none), for the example given.